### PR TITLE
Fix stream has already been operated upon or closed

### PR DESCRIPTION
**Pull request by hamadodene**
issue #456

### CODE REVIEW

File: carapace-server/src/main/java/org/carapaceproxy/server/config/SSLCertificateConfiguration.java [hunk 1]
```diff
@@ -93,10 +93,13 @@ public boolean isMoreSpecific(SSLCertificateConfiguration other) {
...
-        final var otherNames = other.getNames().stream().map(CertificatesUtils::removeWildcard);
-        for (var n: getNames()) {
+
+        final var otherNamesList = other.getNames().stream().map(CertificatesUtils::removeWildcard)
+                .collect(Collectors.toList());
+
+        for (var n : getNames()) {
             final var name = CertificatesUtils.removeWildcard(n);
-            if (otherNames.anyMatch(on -> name.length() > on.length())) {
+            if (otherNamesList.stream().anyMatch(on -> name.length() > on.length())) {
```
Comment (NiccoMlt) [inline carapace-server/src/main/java/org/carapaceproxy/server/config/SSLCertificateConfiguration.java:102]: Here couldn't we just store the max length of the name in otherNamesList and check against it, instead cycle through `otherNamesList` each time?

Comment (hamadodene) [reply carapace-server/src/main/java/org/carapaceproxy/server/config/SSLCertificateConfiguration.java:102]: yes, make sense @NiccoMlt

**Review by NiccoMlt — APPROVED** [review]
Looks good to me

File: carapace-server/src/main/java/org/carapaceproxy/core/Listeners.java [hunk 2]
```diff
@@ -148,6 +149,7 @@ private void stopListener(HostPort hostport) throws InterruptedException {
         ListeningChannel channel = listeningChannels.remove(hostport);
         if (channel != null) {
             channel.channel.disposeNow(Duration.ofSeconds(10));
+            FutureMono.from(channel.getConfig().getGroup().close()).block(Duration.ofSeconds(30));
```
Comment (NiccoMlt) [inline carapace-server/src/main/java/org/carapaceproxy/core/Listeners.java:152]: I'm not sure if 30s is too much here.

Maybe we can go on with this, and change it or make it configurable if it becomes a problem

**Review by NiccoMlt — APPROVED** [review]
LGTM

File: carapace-server/src/main/java/org/carapaceproxy/core/Listeners.java (hunk 2 above)
Comment (hamadodene) [reply carapace-server/src/main/java/org/carapaceproxy/core/Listeners.java:152]: Maybe we can reduce to 10s?

Comment (NiccoMlt) [reply carapace-server/src/main/java/org/carapaceproxy/core/Listeners.java:152]: It might be an improvement, for sure

**Review by NiccoMlt — APPROVED** [review]

**Pull request merged**